set.mm

```diff
@@ -662596,6 +663406,1223 @@ closed under the multiplication ( ' X ' ) of a finite number of
+      IVJVHVIVKVLVHVI $.
+  $}
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+                           Inferior limit (lim inf)
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c liminf $.
+
+  $( Extend class notation to include the liminf function. $)
+  clsi $a class liminf $.
+
+  $d x k $.
+  $( Define the inferior limit of an infinite sequence of extended real
+     numbers.  (Contributed by GS, 2-Jan-2022.) $)
+  df-liminf $a |- liminf = ( x e. _V |-> sup ( ran ( k e. RR |-> inf ( ( ( x "
+    ( k [,) +oo ) ) i^i RR* ) , RR* , < ) ) , RR* , < ) ) $.
+
+  ${
+    $d A i k $.  $d B i $.  $d F i j $.  $d F i k $.  $d i ph $.
+    limsuplt2.1 $e |- ( ph -> B C_ RR ) $.
+    limsuplt2.2 $e |- ( ph -> F : B --> RR* ) $.
+    limsuplt2.3 $e |- ( ph -> A e. RR* ) $.
+    $( The defining property of the superior limit.  (Contributed by Glauco
+       Siliprandi, 2-Jan-2022.) $)
+    limsuplt2 $p |- ( ph -> ( ( limsup ` F ) < A <-> E. k e. RR sup ( ( ( F " (
+       k [,) +oo ) ) i^i RR* ) , RR* , < ) < A ) ) $=
+      ( vi vj clt wbr cv cr cpnf cico co cima cxr cin clsp cfv csup cmpt wss wf
+      wrex wcel wb eqid limsuplt syl3anc wa cvv wceq oveq1 imaeq2d ineq1d simpr
+      supeq1d xrltso supex a1i fvmptd3 breq1d rexbidva cbvrexv 3bitrd ) AEUAUBB
+      KLZIMZJNEJMZOPQZRZSTZSKUCZUDZUBZBKLZINUGZEVJOPQZRZSTZSKUCZBKLZINUGZEDMZOP
+      QZRZSTZSKUCZBKLZDNUGZACNUECSEUFBSUHVIVSUIFGHBCIJEVPVPUJZUKULAVRWDINAVJNUH
+      ZUMZVQWCBKWOJVJVOWCNVPUNWMVKVJUOZSVNWBKWPVMWASWPVLVTEVKVJOPUPUQURUTAWNUSW
+      CUNUHWOSWBKVAVBVCVDVEVFWEWLUIAWDWKIDNVJWFUOZWCWJBKWQSWBWIKWQWAWHSWQVTWGEV
+      JWFOPUPUQURUTVEVGVCVH $.
+  $}
+
+  ${
+    $d A w x y z $.  $d B w x y z $.  $d F x $.
+    $( Ordering property of the superior limit function.  (Contributed by
+       Glauco Siliprandi, 2-Jan-2022.) $)
+    liminfgord $p |- ( ( A e. RR /\ B e. RR /\ A <_ B ) -> inf ( ( ( F " ( A
+       [,) +oo ) ) i^i RR* ) , RR* , < ) <_ inf ( ( ( F " ( B [,) +oo ) ) i^i
+       RR* ) , RR* , < ) ) $=
+      ( vx vy vz vw cr wcel cle wbr cpnf cico co cima cxr cin clt cinf wss wral
+      w3a cv wa inss2 a1i rexr 3ad2ant1 simp3 df-ico xrletr ixxss1 imass2 ssrin
+      syl2anc sselda infxrlb ralrimiva wb infxrcl ax-mp infxrgelb mp2an sylibr
+      3syl ) AHIZBHIZABJKZUBZCALMNZOZPQZPRSZDUCZJKZDCBLMNZOZPQZUAZVMVRPRSJKZVIV
+      ODVRVIVNVRIUDZVLPTZVNVLIVOWBWAVKPUEZUFVIVRVLVNVIVPVJTZVQVKTVRVLTVIAPIZVHW
+      DVFVGWEVHAUGUHVFVGVHUIDEFGABLMJRJMJDEFUJZWFABGUCUKULUOVPVJCUMVQVKPUNVEUPV
+      LVNUQUOURVRPTVMPIZVTVSUSVQPUEWBWGWCVLUTVADVRVMVBVCVD $.
+  $}
+
+  ${
+    $d F k $.
+    limsupvald.1 $e |- ( ph -> F e. V ) $.
+    limsupvald.2 $e |- G = ( k e. RR |-> sup ( ( ( F " ( k [,) +oo ) ) i^i RR*
+       ) , RR* , < ) ) $.
+    $( The superior limit of an infinite sequence ` F ` of extended real
+       numbers, which is the infimum of the set of suprema of all upper
```

Comment:
I believe I have used the same comment as ~ limsupval (for consistency)

To say the truth, I was not so happy about neither the "sequence" nor the "subsequence" terms, because the definition would be "meaningful" for any set: in particular, for functions with dense domain in RR (or defined on all RR).
But the "standard" use will be with sequences, that's why I adapted Norman's comment.

Reply:
last commit:
- I've replaced an infinite sequence --> a sequence
- , which is --> is
- now it is    ... of the set of suprema of all restrictions of ` F ` to an upperset of reals